### PR TITLE
spring-07

### PR DESCRIPTION
**Pull request by AndreiYazadzhi**

### CODE REVIEW

File: src/main/java/com/hibernate/controller/OrderController.java [hunk 1]
```diff
@@ -35,13 +37,21 @@ public OrderController(OrderService orderService,
     }
 
     @PostMapping("/complete")
-    public void completeOrder(@RequestParam Long userId) {
-        orderService.completeOrder(shoppingCartService.getByUser(userService.get(userId)));
+    public void completeOrder(Authentication authentication) {
+        UserDetails userDetails = (UserDetails) authentication.getPrincipal();
+        String email = userDetails.getUsername();
+        User user = userService.getByEmail(email).orElseThrow(()
+                -> new RuntimeException("No user found with such email " + email));
```
Comment (semyonich) [inline src/main/java/com/hibernate/controller/OrderController.java]: If user is authenticated-it exists in our DB

File: src/main/java/com/hibernate/controller/OrderController.java [hunk 2]
```diff
@@ -35,13 +37,21 @@ public OrderController(OrderService orderService,
...
     }
 
     @GetMapping
-    public List<OrderResponseDto> getOrderHistory(@RequestParam Long userId) {
-        return orderService.getOrdersHistory(userService.get(userId)).stream()
+    public List<OrderResponseDto> getOrderHistory(Authentication authentication) {
+        UserDetails userDetails = (UserDetails) authentication.getPrincipal();
+        String email = userDetails.getUsername();
+        User user = userService.getByEmail(email).orElseThrow(()
+                -> new RuntimeException("There is no such user with email " + email));
```
Comment (semyonich) [inline src/main/java/com/hibernate/controller/OrderController.java]: same

File: src/main/java/com/hibernate/controller/ShoppingCartController.java [hunk 3]
```diff
@@ -34,14 +37,20 @@ public ShoppingCartController(MovieSessionService movieSessionService,
...
     @PostMapping("/movie-sessions")
-    public void addMovieSession(@RequestParam Long userId,
-                                                   @RequestParam Long movieSessionId) {
-        shoppingCartService.addSession(movieSessionService
-                .get(movieSessionId), userService.get(userId));
+    public void addMovieSession(Authentication authentication, @RequestParam Long movieSessionId) {
+        UserDetails userDetails = (UserDetails) authentication.getPrincipal();
+        String email = userDetails.getUsername();
+        User user = userService.getByEmail(email).orElseThrow(()
+                -> new RuntimeException("There is no such user with email " + email + "."));
```
Comment (semyonich) [inline src/main/java/com/hibernate/controller/ShoppingCartController.java]: same

File: src/main/java/com/hibernate/controller/ShoppingCartController.java [hunk 4]
```diff
@@ -34,14 +37,20 @@ public ShoppingCartController(MovieSessionService movieSessionService,
...
     }
 
     @GetMapping("/by-user")
-    public ShoppingCartResponseDto getByUser(@RequestParam Long userId) {
-        return responseMapper.toDto(shoppingCartService.getByUser(userService.get(userId)));
+    public ShoppingCartResponseDto getByUser(Authentication authentication) {
+        UserDetails userDetails = (UserDetails) authentication.getPrincipal();
+        String email = userDetails.getUsername();
+        User user = userService.getByEmail(email).orElseThrow(()
+                -> new RuntimeException("There is no such user with email " + email + "."));
```
Comment (semyonich) [inline src/main/java/com/hibernate/controller/ShoppingCartController.java]: same

File: src/main/java/com/hibernate/lib/validators/PasswordValidator.java [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+public class PasswordValidator implements
+        ConstraintValidator<PasswordValidation, Object> {
+
+    @Override
+    public boolean isValid(Object value, ConstraintValidatorContext constraintValidatorContext) {
+        UserRequestDto user = (UserRequestDto) value;
+        return user.getPassword().equals(user.getRepeatPassword());
```
Comment (semyonich) [inline src/main/java/com/hibernate/lib/validators/PasswordValidator.java:18]: What if user.password=null ?

File: src/main/java/com/hibernate/model/dto/request/UserRequestDto.java [hunk 6]
```diff
@@ -1,8 +1,29 @@
...
+import com.hibernate.lib.annotations.EmailValidation;
+import com.hibernate.lib.annotations.PasswordValidation;
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotNull;
+
+@PasswordValidation(message = "Passwords are not equal")
 public class UserRequestDto {
+    @NotNull
+    @EmailValidation
+    @Min(8)
```
Comment (semyonich) [inline src/main/java/com/hibernate/model/dto/request/UserRequestDto.java]: you have already this check in validator

**Review by semyonich — CHANGES_REQUESTED** [review]

File: src/main/java/com/hibernate/lib/validators/PasswordValidator.java [hunk 7]
```diff
@@ -11,6 +11,10 @@
     @Override
     public boolean isValid(Object value, ConstraintValidatorContext constraintValidatorContext) {
         UserRequestDto user = (UserRequestDto) value;
+        if (user.getPassword() == null
+                || user.getRepeatPassword() == null) {
+            return false;
+        }
         return user.getPassword().equals(user.getRepeatPassword());
```
Comment (semyonich) [inline src/main/java/com/hibernate/lib/validators/PasswordValidator.java]:
```suggestion
        if (user.getPassword() == null)                
            return false;
        }
        return  user.getPassword().equals(user.getRepeatPassword());
```

**Review by semyonich — APPROVED** [review]

**Review by boroda4436 — APPROVED** [review]

**Pull request merged**